capabilities/version-control.md

```diff
@@ -1,28 +1,109 @@
+
+Without a robust version control strategy, teams are left vulnerable to disruptions in productivity and system reliability. They may struggle to reproduce environments, leading to prolonged downtime and inefficiency. Auditability is also compromised, making regulatory compliance difficult to demonstrate. Defect response suffers, as teams lack the ability to swiftly restore a known working state. 
+
+By prioritizing effective version control practices, organizations can mitigate these risks and ensure seamless, reliable, and compliant software delivery.
+
+## Nuances
+
+This section outlines common pitfalls, challenges, or limitations teams commonly encounter when applying this capability. The goal here is not to discourage you. Rather, the goal is to arm you with the appropriate context so that you can make an informed decision about when and how to implement the capability with your teams.
+
+### Deterministic Builds vs. Up-to-Date Dependencies
+
+Version control systems often face a trade-off between deterministic builds and up-to-date dependencies. Some developers prefer specifying dependencies to use the latest versions at build time for enhanced security. However, this compromises build determinism, making it challenging to reproduce identical builds. Tools like [Dependabot](https://github.com/dependabot) can mitigate this issue by automatically creating pull requests to update dependencies, separating dependency updates from the build process.
+
+### Versioning Non-Code
+
+Version control extends beyond code to configuration, data, infrastructure, dependencies, and documentation. Establishing a version control strategy for these categories ensures a deterministic build process. This comprehensive approach enhances reproducibility, traceability, and disaster recovery. Working in a system with a deterministic build process simplifies troubleshooting, deployment, and maintenance.
+
+### Working With Large Assets
+
+Traditional version control systems like Git struggle with large file sizes. Storing large files in these systems can lead to inefficiencies, as entire file copies are stored between versions instead of diffs. This limitation necessitates specialized systems or strategies for managing large assets, ensuring efficient storage and versioning.
+
+### Security, Compliance, and Access Control
+
+Strict version control helps teams identify security vulnerabilities and exploits that have been incorporated into the system. However, version control systems must comply with regulatory requirements like GDPR, HIPAA, or PCI-DSS, which adds complexity. Storing sensitive data in these broadly accessible repositories poses risks. Effective access control, encryption, and auditing mechanisms are crucial to mitigate these challenges.
```

Comment:
I’m having trouble connecting these two sentences: Strict version control helps teams identify security vulnerabilities and exploits that have been incorporated into the system. However, version control systems must comply with regulatory requirements like GDPR..." 

Do we mean that in order to identify security vulnerabilities, a certain amount of sensisitve data must be accessed?

Reply:
Good call-out. I'm making an assumption that readers know the underlying "how" without going into detail. That's a bad assumption to make.

Certain versions of software can be flagged as vulnerable by the community [CVE for example](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures). Having good version control makes it easy to work with this data and make upgrades to the version of vulnerable dependencies (ie, use a more recent version that doesn't suffer from the same vulnerability).

Reply:
See my edits in 12/30 commit. I'm not 100% sure I've articulated your clarification accurately.